sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -188,19 +188,11 @@ class UnivocityParser(
     }
   }
 
-  private val doParse = if (requiredSchema.nonEmpty) {
-    (input: String) => convert(tokenizer.parseLine(input))
-  } else {
-    // If `columnPruning` enabled and partition attributes scanned only,
-    // `schema` gets empty.
-    (_: String) => InternalRow.empty
```

Comment:
Does it still prune columns when `spark.sql.csv.parser.columnPruning.enabled` enabled? In this way we will still parse input CSV.

Reply:
yes I tested with  the below data

#fruit.csv
apple,red,1,3
banana,yellow,2,4
orange

![image](https://user-images.githubusercontent.com/35216143/65100440-a45ff100-d9e5-11e9-8464-8d6e4f8c491e.png)
![image](https://user-images.githubusercontent.com/35216143/65100974-e38e4200-d9e5-11e9-8271-d8a6ceeab360.png)

Reply:
No, I mean internally it will skip parsing phrase which will bring performance regression. When `spark.sql.csv.parser.columnPruning.enabled` is disabled, seems it's right to parse; however, when it's enabled, we should skip the parse logic.

Reply:
Previously, it skipped parsing logic when there are no columns are selected, which seems correct.

Reply:
ahh!!!, you are correct, we should only parse when `spark.sql.csv.parser.columnPruning.enabled` disabled or `requiredSchema.nonEmpty`.

I have updated the code